recipes/cgns/all/patches/fix_find_hdf5.patch

```diff
@@ -0,0 +1,109 @@
+-      else (HDF5_static_C_FOUND)
+-        set (LINK_LIBS ${LINK_LIBS} ${HDF5_C_STATIC_LIBRARY})
+-      endif (BUILD_SHARED_LIBS AND HDF5_shared_C_FOUND)
+-      set_property (TARGET h5dump PROPERTY IMPORTED_LOCATION "${HDF5_TOOLS_DIR}/h5dump")
+-    endif (NOT HDF5_static_C_FOUND AND NOT HDF5_shared_C_FOUND)
+-    set (HDF5_DUMP_EXECUTABLE $<TARGET_FILE:h5dump>)
+-
+-    set (HDF5_HAVE_H5PUBCONF_H 1)
+-    set (HDF5_HAVE_HDF5 1)
+-    set (HDF5_HDF5_HEADER "h5pubconf.h")
+-    set (HDF5_INCLUDE_DIR_GEN ${HDF5_INCLUDE_DIR})
+-    message (STATUS "HDF5-${HDF5_VERSION_STRING} found: INC=${HDF5_INCLUDE_DIR} TOOLS=${HDF5_TOOLS_DIR}")
+-  else (HDF5_FOUND)
+-    find_package (HDF5) # Legacy find
+-#Legacy find_package does not set HDF5_TOOLS_DIR, so we set it here
+-    set(HDF5_TOOLS_DIR ${HDF5_LIBRARY_DIRS}/../bin)
+-#Legacy find_package does not set HDF5_BUILD_SHARED_LIBS, so we set it here
+-    if (CGNS_BUILD_SHARED AND EXISTS "${HDF5_LIBRARY_DIRS}/libhdf5${CMAKE_SHARED_LIBRARY_SUFFIX}")
+-      set (HDF5_BUILD_SHARED_LIBS 1)
+-      add_definitions (-DH5_BUILT_AS_DYNAMIC_LIB)
+-    else ()
+-      set (HDF5_BUILD_SHARED_LIBS 0)
+-      add_definitions (-DH5_BUILT_AS_STATIC_LIB)
+-    endif ()
+-    set (LINK_LIBS ${LINK_LIBS} ${HDF5_LIBRARIES})
+-
+-    add_executable (h5dump IMPORTED)
+-    set_property (TARGET h5dump PROPERTY IMPORTED_LOCATION "${HDF5_TOOLS_DIR}/h5dump")
+-    set (HDF5_DUMP_EXECUTABLE $<TARGET_FILE:h5dump>)
+-  endif (HDF5_FOUND)
+-  set (HDF5_PACKAGE_NAME ${SEARCH_PACKAGE_NAME})
+-
+-  if (HDF5_FOUND)
+-    set (HDF5_HAVE_H5PUBCONF_H 1)
+-    set (HDF5_HAVE_HDF5 1)
+-    set (HDF5_HDF5_HEADER "h5pubconf.h")
+-    set (HDF5_INCLUDE_DIR_GEN ${HDF5_INCLUDE_DIR})
+-    message (STATUS "HDF5-${HDF5_VERSION_STRING} found: INC=${HDF5_INCLUDE_DIR} TOOLS=${HDF5_TOOLS_DIR} HDF5_BUILD_SHARED_LIBS=${HDF5_BUILD_SHARED_LIBS}")
+-    INCLUDE_DIRECTORIES (${HDF5_INCLUDE_DIR})
+-    set (HDF5_LIBRARY ${LINK_LIBS})
+-    message (STATUS "HDF5 link libs: ${LINK_LIBS}")
+-  else (HDF5_FOUND)
+-    message (FATAL_ERROR " HDF5 was specified but was not found")
+-  endif (HDF5_FOUND)
+-
+-  set(HDF5_NEED_ZLIB "OFF" CACHE BOOL "Does the HDF5 library require linking to zlib?")
+-  if(HDF5_NEED_ZLIB)
+-    find_library(ZLIB_LIBRARY z)
+-    mark_as_advanced(CLEAR ZLIB_LIBRARY)
+-  else (HDF5_NEED_ZLIB)
+-    mark_as_advanced(FORCE ZLIB_LIBRARY)
+-  endif(HDF5_NEED_ZLIB)
+-
+-  set(HDF5_NEED_SZIP "OFF" CACHE BOOL "Does the HDF5 library require linking to zlib?")
+-  if (HDF5_NEED_SZIP)
+-    find_library(SZIP_LIBRARY szip)
+-    mark_as_advanced(CLEAR SZIP_LIBRARY)
+-  else (HDF5_NEED_SZIP)
+-    mark_as_advanced(FORCE SZIP_LIBRARY)
+-  endif (HDF5_NEED_SZIP)
+-
+-  # Check if HDF5 version is 1.8 or greater
+-  if (HDF5_VERSION VERSION_LESS "1.8.0")
+-    message (FATAL_ERROR "HDF5 version must be greater than 1.8.0")
+-  endif (HDF5_VERSION VERSION_LESS "1.8.0")
++  INCLUDE_DIRECTORIES (${CONAN_INCLUDE_DIRS})
++  set (HDF5_LIBRARY ${CONAN_LIBS})
```

Comment:
I am not crazy for this patch... there's a ton of logic in this script configuring flags and adding extra headers

It's better to link exclusively to the one target... we'll need to update the patch however
```suggestion
+  set (HDF5_LIBRARY CONAN_PKG::hdf5)
```

Reply:
I know, I am not a big fan of this patch as well.
But considering that the FindHDF5 module of cmake uses the hdf5 compiler, I did not see another solution.
(https://cmake.org/cmake/help/latest/module/FindHDF5.html)

I am now linking against the target CONAN_PKG::hdf5 as you suggested.
But still I need to call the global include_directories. Not every target that actually uses the hdf5 headers links against the hdf5 libs :-(